server/data/data-api/src/main/java/org/apache/james/droplists/api/DropListEntry.java

```diff
@@ -71,6 +76,11 @@ public DropListEntry build() {
             Preconditions.checkArgument(deniedEntity != null && !deniedEntity.isBlank(), "`deniedEntity` must not be null, empty, or blank");
             Preconditions.checkArgument(deniedEntity.length() <= MAXIMUM_DOMAIN_LENGTH,
                 "deniedEntity length should not be longer than %s characters", MAXIMUM_DOMAIN_LENGTH);
+            if (deniedEntityType.equals(DeniedEntityType.DOMAIN)) {
+                deniedEntity = Domain.of(deniedEntity).asString();
+            } else {
+                deniedEntity = new MailAddress(deniedEntity).toString();
+            }
```

Comment:
Generally speaking we can avoid variable reallocation by method extraction.
This is more elegant and less complex.